### PR TITLE
Adds ipa_automountkey module

### LINKED ISSUE #65719: FreeIPA: There is no way to modify Automount Keys
##### SUMMARY

There is no way to modify IPA's Automount Keys. I will be contributing a module in an upcoming PR.
##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

`lib/ansible/modules/identity/ipa`

##### ADDITIONAL INFORMATION

There is no module for modifying IPA's Automount Keys yet.

```
$ ansible --version
ansible 2.9.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/b1czu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.8/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.0 (default, Oct 23 2019, 18:51:26) [GCC 9.2.0]

```

### PR DESCRIPTION
**Pull request by b1czu**
##### SUMMARY

This PR adds an ipa_automountkey module which allows to manage IPA Automount Keys.

Fixes #65719
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

We did not have an ipa_automountkey module, so here's one!

### CODE REVIEW

**Comment by b1czu** [comment]
@Akasurde @Nosmoht @fxfitz can you help me with this failing documentation test? I've submitted  IPA module with similar documentation before and it passed OK. Now, for this one, it complains about ipa_pass argument but it is not in argument_spec or documentation part of my `ipa_automountkey.py` file.

Also local tests run with `ansible-test sanity -v --junit --coverage-check --docker-keep-git --base-branch origin/devel --allow-disabled --require lib/ansible/modules/identity/ipa/ipa_automountkey.py --docker default
` do not throw this error.

Any help in fixing this would be appreciated.

File: lib/ansible/modules/identity/ipa/ipa_automountkey.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+short_description: Manage FreeIPA Automount Keys
+description:
+- Add, delete and modify an IPA Automount Key using IPA API.
+options:
+  state:
+    description: State to ensure
+    required: false
+    type: str
+    default: present
+    choices: ["present", "absent"]
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_automountkey.py]:
```suggestion
    choices: ["absent", "present"]
```

File: lib/ansible/modules/identity/ipa/ipa_automountkey.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    description: Mount information.
+    required: false
+    type: str
+    aliases: ["information"]
+extends_documentation_fragment: ipa.documentation
+version_added: "2.10"
+'''
+
+EXAMPLES = '''
+# Ensure that /exports/nas key in auto.direct map is present
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_automountkey.py]:
```suggestion
- name: Ensure that /exports/nas key in auto.direct map is present
  ipa_automountkey:
```

File: lib/ansible/modules/identity/ipa/ipa_automountkey.py [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    state: present
+    key: /exports/nas
+    mountinformation: '-rw,soft,rsize=8192,wsize=8192 nas-server:/exports/nas/'
+    location: default
+    mapname: auto.direct
+    ipa_host: localhost
+    ipa_user: admin
+    ipa_pass: topsecret
+
+# Ensure that /exports/nas key in auto.direct map is absent
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_automountkey.py]:
```suggestion
- name: Ensure that /exports/nas key in auto.direct map is absent
  ipa_automountkey:
```

File: lib/ansible/modules/identity/ipa/ipa_automountkey.py [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+    ipa_host: localhost
+    ipa_user: admin
+    ipa_pass: topsecret
+'''
+
+RETURN = '''
+automountkey:
+  description: Automount Key as returned by IPA API
+  returned: always
+  type: dict
```
Comment (Akasurde) [inline lib/ansible/modules/identity/ipa/ipa_automountkey.py:80]: Add a sample over here how this dict will look like.

**Review by Akasurde — CHANGES_REQUESTED** [review]
Rest LGTM

**Comment by Akasurde** [comment]
I re-triggered the CI and fixed the issue for above CI failure.

**Comment by b1czu** [comment]
ready_for_review

**Comment by b1czu** [comment]
bot_status

**Pull request closed without merging**